components/calendar/src/iso.rs

```diff
@@ -415,18 +417,15 @@ impl Iso {
     // Lisp code reference: https://github.com/EdReingold/calendar-code2/blob/1ee51ecfaae6f856b0d7de3e36e9042100b4f424/calendar.l#L1191-L1217
     fn iso_year_from_fixed(date: i32) -> i32 {
         // 400 year cycles have 146097 days
-        let n_400 = date / 146097;
-        let date = date % 146097;
+        let (n_400, date) = div_rem_euclid(date, 146097);
 
         // 100 year cycles have 36524 days
-        let n_100 = date / 36524;
-        let date = date % 36524;
+        let (n_100, date) = div_rem_euclid(date, 36524);
```

Comment:
Thought: We don't need the new functions when the dividend is known to be positive, as is the case here. Do you think we should still use `div_rem_euclid` in these cases?

One alternative I considered was to make `div_rem_euclid` return `(i32, u32)`.

Reply:
i think it's a cleaner symmetry here, but in general I agree.